extension-admob/api/admob.script_api

```diff
@@ -0,0 +1,454 @@
+- name: admob
+  type: table
+  desc: Functions and constants for interacting with [Google AdMob APIs](https://developers.google.com/admob)
+
+  members:
+
+#*****************************************************************************************************
+
+  - name: initialize
+    type: function
+    desc: "Initialize the Mobile Ads SDK.
+          Warning: If you need to obtain consent from users in the European Economic Area (EEA), set
+          any request-specific flags, or otherwise take action before loading ads, ensure you do so
+          before initializing the Mobile Ads SDK.
+
+          Original docs
+          [Android](https://developers.google.com/admob/android/quick-start#initialize_the_mobile_ads_sdk),
+          [iOS](https://developers.google.com/admob/ios/quick-start#initialize_the_mobile_ads_sdk)"
+
+#*****************************************************************************************************
+
+  - name: set_callback
+    type: function
+    desc: Sets a callback function for receiving events from the SDK. Call `admob.set_callback(nil)`
+          to remove callback
+
+    parameters:
+    - name: callback
+      type: function
+      desc: Callback function that is executed on any event in the SDK.
+
+      parameters:
+        - name: self
+          type: object
+          desc: The calling script instance
+
+        - name: message_id
+          type: number
+          desc: "One of message types:
+               `admob.MSG_INITIALIZATION` initialization,
+               `admob.MSG_INTERSTITIAL` message from Interstitial ad unit,
+               `admob.MSG_REWARDED` message from Rewarded ad unit,
+               `admob.MSG_BANNER` message from Banner ad unit"
+
+        - name: message
+          type: table
+          desc: A table holding the data
+          fields:
+              - name: event
+                type: number
+                desc: "One of event types:
+                      `admod.EVENT_CLOSED`,
+                      `admod.EVENT_FAILED_TO_SHOW`,
+                      `admod.EVENT_OPENING`,
+                      `admod.EVENT_FAILED_TO_LOAD`,
+                      `admod.EVENT_LOADED`,
+                      `admod.EVENT_NOT_LOADED`,
+                      `admod.EVENT_EARNED_REWARD`,
+                      `admod.EVENT_COMPLETE`,
+                      `admod.EVENT_CLICKED`,
+                      `admod.EVENT_DESTROYED`"
```

Comment:
there is one more event in code `EVENT_JSON_ERROR`, that would be nice to add this even in doc and in example

Reply:
Probably 'internal' mark slightly confused me.. But I fully agree that event will be useful for user-specific logging and issues reporting.